addons/l10n_ch/tests/test_swissqr.py

```diff
@@ -164,8 +166,9 @@ def swissqr_generated(self, invoice, ref_type='NON'):
     def test_swissQR_missing_bank(self):
         # Let us test the generation of a SwissQR for an invoice, first by showing an
         # QR is included in the invoice is only generated when Odoo has all the data it needs.
-        self.invoice1.action_post()
-        self.swissqr_not_generated(self.invoice1)
+        with self.assertRaises(UserError):
```

Comment:
The best practice is: `with self.assertRaises(UserError), self.cr.savepoint():` Because in case of error, the cursor is broken and the transaction is not rollback meaning everything added after will be broken